### ISSUE TITLE
In Python 3.4, unittest's reporter changed unexpectedSuccesses to mean a suite failure

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7885]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#7885|
|Type|defect|
|Created|2015-05-04 11:22:09Z|
|Branch|https://github.com/twisted/twisted/tree/unittest34changes-7885|

See https://docs.python.org/3/library/unittest.html#unittest.TestResult.wasSuccessful

This causes the 3.4 and 3.5 builders to fail.

We should figure out what behaviour is correct, and fix our tests to want that.

<details><summary>Searchable metadata</summary>

```
trac-id__7885 7885
type__defect defect
reporter__hawkowl hawkowl
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_unittest34changes_7885 branches/unittest34changes-7885
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__trial trial
keywords__ 
time__1430738529327065 1430738529327065
changetime__1430890250194331 1430890250194331
version__None None
owner__hawkowl hawkowl
cc__jml
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_2"></a>|@glyph commented|
|-|-|

This is a bad change. We should probably report a bug against Python.

In the meanwhile though, perhaps this is an opportunity to simply eliminate all of our .todo'd tests and turn them into tickets?  Probably not all at once, though.  `.todo` is a pretty silly feature for us to be using anyway.

As far as testing Trial itself, just change the tests to have a version check in the test itself, since apparently the meaning changed in Python.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

(In [44658]) Branching to unittest34changes-7885.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

I think the best thing to do is introduce the old Python functionality into our test runner, because then it won't cause weird failures when Twisted applications go from py33 to py34.

Builders spun (please especially see the unsupported builders at https://buildbot.twistedmatrix.com/boxes-unsupported?branch=/branches/unittest34changes-7885 ), please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban commented|
|-|-|

I think that all .todo tests were already removed and converted into ticket, I have worked on this. I have also worked at updating the docs to talk about not committing todo tests into trunk.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_6"></a>|@glyph commented|
|-|-|

Replying to [adiroiban](#note_5):
> I think that all .todo tests were already removed and converted into ticket, I have worked on this. I have also worked at updating the docs to talk about not committing todo tests into trunk.

Thanks for doing this, adi!  I guess we're moving faster than I'm used to now, which is fantastic :-D.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_7"></a>|@glyph set owner to @hawkowl|
|-|-|

Thanks, hawkie!

The docstring for `wasSuccesful` should detail the nature of the change in 3.4, and the behavior which it actually has.  Other than that, I think this looks good to land, so please add a little prose and do so.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_8"></a>|@hawkowl set status to `closed`|
|-|-|

(In [44667]) Merge unittest34changes-7885: Make the pre-Python3.3 unittest reporter wasSuccessful behaviour concrete

Author: hawkowl
Reviewer: glyph
Fixes: [#7885](https://github.com/twisted/twisted/issues/7885)
